client-spark/common/src/main/java/org/apache/spark/shuffle/writer/AddBlockEvent.java

```diff
@@ -21,12 +21,14 @@
 import java.util.List;
 
 import org.apache.uniffle.common.ShuffleBlockInfo;
+import org.apache.uniffle.common.function.TupleConsumer;
 
 public class AddBlockEvent {
 
   private String taskId;
   private List<ShuffleBlockInfo> shuffleDataInfoList;
   private List<Runnable> processedCallbackChain;
+  private TupleConsumer<ShuffleBlockInfo, Boolean> blockSentCallback;
```

Comment:
Could we give a more unify design between processedCallbackChain and blockSentCallback?

Reply:
I will remove processedCallbackChain in the future, it looks unnecessary.

Reply:
From current usage, needn't

Reply:
```
interface BlockFailureCallback {
onBlockFailure(ShuffleBlock block);
}

interface BlockSuccessCallback {
onBlockSuccess(ShuffleBlock block);
}

interface BlockCompletionCallback {
onBlockCompletion(ShuffleBlock block);
}
```